Regression with ruff 0.0.42 on relative exclude files

Hi again,

apologies for creating so many issues and for sometimes missing to catch up, because the pace is so fast. I hope you still appreciate my feedback.

After adjusting our exclude pattern recently as outlined at https://github.com/charliermarsh/ruff/issues/203#issuecomment-1249055408 and https://github.com/jpmens/mqttwarn/commit/fad50af912e, we just had to downgrade to ruff 0.0.40 (https://github.com/jpmens/mqttwarn/commit/041a3f425d) to make it work again.

Currently, on 0.0.42, `ruff` would croak on both configuration variants of `extend-exclude`. There is probably another thing amongst the recent changes I am missing here?

```
tests/etc/functions_bad.py:0:0: E999 SyntaxError: unexpected indent at line 6 column 5
```
```toml
extend-exclude = [
  "./tests/etc/functions_bad.py"
]
```
```toml
extend-exclude = [
  "tests/etc/functions_bad.py"
]
```

Keep up the spirit and with kind regards,
Andreas.

P.S.: Just to make it clear: [`tests/etc/functions_bad.py`](https://github.com/jpmens/mqttwarn/blob/0.29.1/tests/etc/functions_bad.py) _intentionally_ has bad formatting, because we use it as a canary for running our plugin machinery on bad input files. That is why we need to make `ruff` skip it.

--- a/src/fs.rs
+++ b/src/fs.rs
@@ -66,7 +66,7 @@ pub fn iter_python_files<'a>(
     let exclude_simple = exclude
         .iter()
         .all(|pattern| matches!(pattern, FilePattern::Simple(_)));
-    let extend_exclude_simple = exclude
+    let extend_exclude_simple = extend_exclude
         .iter()
         .all(|pattern| matches!(pattern, FilePattern::Simple(_)));
 
@@ -114,9 +114,6 @@ pub fn iter_python_files<'a>(
 
 /// Convert any path to an absolute path (based on the current working directory).
 pub fn normalize_path(path: &Path) -> PathBuf {
-    if path == Path::new(".") || path == Path::new("..") {
-        return path.to_path_buf();
-    }
     if let Ok(path) = path.absolutize() {
         return path.to_path_buf();
     }
@@ -125,9 +122,6 @@ pub fn normalize_path(path: &Path) -> PathBuf {
 
 /// Convert any path to an absolute path (based on the specified project root).
 pub fn normalize_path_to(path: &Path, project_root: &Path) -> PathBuf {
-    if path == Path::new(".") || path == Path::new("..") {
-        return path.to_path_buf();
-    }
     if let Ok(path) = path.absolutize_from(project_root) {
         return path.to_path_buf();
     }

--- a/src/pyproject.rs
+++ b/src/pyproject.rs
@@ -1,9 +1,8 @@
-use std::ops::Deref;
 use std::path::{Path, PathBuf};
 
 use anyhow::Result;
 use common_path::common_path_all;
-use path_absolutize::path_dedot;
+use path_absolutize::Absolutize;
 use serde::Deserialize;
 
 use crate::checks::CheckCode;
@@ -78,8 +77,10 @@ fn find_user_pyproject_toml() -> Option<PathBuf> {
 }
 
 pub fn find_project_root(sources: &[PathBuf]) -> Option<PathBuf> {
-    let cwd = path_dedot::CWD.deref();
-    let absolute_sources: Vec<PathBuf> = sources.iter().map(|source| cwd.join(source)).collect();
+    let absolute_sources: Vec<PathBuf> = sources
+        .iter()
+        .flat_map(|source| source.absolutize().map(|path| path.to_path_buf()))
+        .collect();
     if let Some(prefix) = common_path_all(absolute_sources.iter().map(PathBuf::as_path)) {
         for directory in prefix.ancestors() {
             if directory.join(".git").is_dir() {

